### PR TITLE
attestation orgpolicy repo changes

### PR DESCRIPTION
**Pull request by arvindsingh-cloud**
## Description
</br>
 Brief description of the requested change.

## Checklist
- [ ] I have read the instructions mentioned in [Contribute to Code](/Contributing.md).
- [ ] I have read and understood the criteria described under [submitting changes](/Contributing.md#submitting-changes). 
- [ ] The title of the PR clearly describes the intent of the PR.
- [ ] This PR does not introduce any breaking changes to the code.

### CODE REVIEW

File: src/AzSK.ADO/Framework/Managers/ControlStateExtension.ps1 [hunk 1]
```diff
@@ -465,13 +474,18 @@ class ControlStateExtension
...
 						#Validate if Attestation repo is available in policy project
+						$attestationRepo = [Constants]::AttestationRepo;
 						try 
 						{
 							$rmContext = [ContextHelper]::GetCurrentContext();
 							$user = "";
 							$base64AuthInfo = [Convert]::ToBase64String([Text.Encoding]::ASCII.GetBytes(("{0}:{1}" -f $user,$rmContext.AccessToken)))
 						
-							$uri = "https://dev.azure.com/{0}/{1}/_apis/git/repositories/{2}/refs?api-version=5.0" -f $this.SubscriptionContext.subscriptionid, $projectName, [Constants]::AttestationRepo 
+							if ([Helpers]::CheckMember($this.ControlSettings,"AttestationRepo")) {
```
Comment (abhaydaga) [inline src/AzSK.ADO/Framework/Managers/ControlStateExtension.ps1:484]: add comment if this variable is empty in COntrolSettings.

File: src/AzSK.ADO/Framework/Managers/ControlStateExtension.ps1 [hunk 2]
```diff
@@ -366,33 +366,41 @@ class ControlStateExtension
 		$fileName = $FullName.split('\')[-1];
 
 		$projectName = $this.GetProject();
+		$attestationRepo = [Constants]::AttestationRepo;
```
Comment (abhaydaga) [inline src/AzSK.ADO/Framework/Managers/ControlStateExtension.ps1:369]: add comment if this variable is empty in COntrolSettings.

File: src/AzSK.ADO/Framework/Managers/ControlStateExtension.ps1 [hunk 3]
```diff
@@ -366,33 +366,41 @@ class ControlStateExtension
...
 		$user = "";
 		$base64AuthInfo = [Convert]::ToBase64String([Text.Encoding]::ASCII.GetBytes(("{0}:{1}" -f $user,$rmContext.AccessToken)))
 	   
-		$uri = "https://dev.azure.com/{0}/{1}/_apis/git/repositories/{2}/refs?api-version=5.0" -f $this.SubscriptionContext.subscriptionid, $projectName, [Constants]::AttestationRepo 
+		$uri = "https://dev.azure.com/{0}/{1}/_apis/git/repositories/{2}/refs?api-version=5.0" -f $this.SubscriptionContext.subscriptionid, $projectName, $attestationRepo 
         try {
 		$webRequest = Invoke-RestMethod -Uri $uri -Method Get -ContentType "application/json" -Headers @{Authorization=("Basic {0}" -f $base64AuthInfo)}
-		$branchId = ($webRequest.value | where {$_.name -eq 'refs/heads/master'}).ObjectId
+		$branchName = [Constants]::AttestationDefaultBranch;
+		if ([Helpers]::CheckMember($this.ControlSettings,"AttestationBranch")) {
```
Comment (abhaydaga) [inline src/AzSK.ADO/Framework/Managers/ControlStateExtension.ps1:382]: add comment if this variable is empty in COntrolSettings.

File: src/AzSK.ADO/Framework/Managers/ControlStateExtension.ps1 [hunk 4]
```diff
@@ -567,7 +581,10 @@ class ControlStateExtension
 	[PSObject] GetRepoFileContent($fileName)
 	{
 		$projectName = $this.GetProject();
-		$branchName =  [Constants]::AttestationBranch
+		$branchName =  [Constants]::AttestationDefaultBranch
+		if ([Helpers]::CheckMember($this.ControlSettings,"AttestationBranch")) {
```
Comment (abhaydaga) [inline src/AzSK.ADO/Framework/Managers/ControlStateExtension.ps1:585]: add comment if this variable is empty in COntrolSettings.

File: src/AzSK.ADO/Framework/Managers/ControlStateExtension.ps1 [hunk 5]
```diff
@@ -577,7 +594,11 @@ class ControlStateExtension
 		
 		try
 		{
-		   $uri = [Constants]::GetAttRepoStorageUri -f $this.SubscriptionContext.subscriptionid, $projectName, [Constants]::AttestationRepo, $fileName, $branchName 
+			$attestationRepo = [Constants]::AttestationRepo;
+			if ([Helpers]::CheckMember($this.ControlSettings,"AttestationRepo")) {
```
Comment (abhaydaga) [inline src/AzSK.ADO/Framework/Managers/ControlStateExtension.ps1:598]: add comment if this variable is empty in COntrolSettings.

File: src/AzSK.ADO/Framework/Managers/ControlStateExtension.ps1 [hunk 6]
```diff
@@ -608,26 +629,35 @@ class ControlStateExtension
 		}
 	}
 
-	[void] RemoveExtStorageContent($fileName)
+	[void] RemoveAttestationData($fileName)
 	{
 		$projectName = $this.GetProject();
 		$fileName = $this.CreatePath($fileName);
+		$attestationRepo = [Constants]::AttestationRepo;
+		if ([Helpers]::CheckMember($this.ControlSettings,"AttestationRepo")) {
```
Comment (abhaydaga) [inline src/AzSK.ADO/Framework/Managers/ControlStateExtension.ps1:637]: add comment if this variable is empty in COntrolSettings.

File: src/AzSK.ADO/Framework/Managers/ControlStateExtension.ps1 (hunk 2 above)
Comment (arvindsingh-cloud) [reply src/AzSK.ADO/Framework/Managers/ControlStateExtension.ps1:369]: added comment

File: src/AzSK.ADO/Framework/Managers/ControlStateExtension.ps1 (hunk 3 above)
Comment (arvindsingh-cloud) [reply src/AzSK.ADO/Framework/Managers/ControlStateExtension.ps1:384]: added comment

File: src/AzSK.ADO/Framework/Managers/ControlStateExtension.ps1 (hunk 1 above)
Comment (arvindsingh-cloud) [reply src/AzSK.ADO/Framework/Managers/ControlStateExtension.ps1:487]: added comment

File: src/AzSK.ADO/Framework/Managers/ControlStateExtension.ps1 (hunk 4 above)
Comment (arvindsingh-cloud) [reply src/AzSK.ADO/Framework/Managers/ControlStateExtension.ps1:589]: added comment

File: src/AzSK.ADO/Framework/Managers/ControlStateExtension.ps1 (hunk 5 above)
Comment (arvindsingh-cloud) [reply src/AzSK.ADO/Framework/Managers/ControlStateExtension.ps1:603]: added comment

File: src/AzSK.ADO/Framework/Managers/ControlStateExtension.ps1 (hunk 6 above)
Comment (arvindsingh-cloud) [reply src/AzSK.ADO/Framework/Managers/ControlStateExtension.ps1:643]: added comment

**Review by abhaydaga — APPROVED** [review]

**Pull request merged**